risc0/zkp/src/prove/soundness.rs

```diff
@@ -0,0 +1,292 @@
+        self.ext_size as f32 * n_columns as f32 * self.trace_domain_size / self.ext_field_size
+    }
+
+    /// (m + 1/2)^7 / (3 * sqrt(ρ)^3) * |D|^2 / |K|
+    fn e_proximity_gap_proven(&self) -> f32 {
+        (M + 0.5).powi(7) / (3.0 * RHO.sqrt().powi(3))
+            * (self.lde_domain_size.powi(2) / self.ext_field_size)
+    }
+
+    /// Conjecture 8.4 [BCIKS21]
+    fn e_proximity_gap_conjectured(&self) -> f32 {
+        let c_1 = 1; // first parameter in Proximity Gaps, Conjecture 8.4
+        let c_2 = 1; // second parameter in Proximity Gaps, Conjecture 8.4
+
+        // 1 / (ηρ)^c_1
+        let first_term = 1.0 / (ETA * RHO).powi(c_1);
+
+        //   (l • n)^c_2 / q
+        // = (n_trace_polys • |D|)^c_2 / ext_field_size
+        let second_term =
+            (self.n_trace_polys * self.lde_domain_size).powi(c_2) / self.ext_field_size;
+
+        first_term * second_term
+    }
+
+    fn e_fri_constant(&self, e_proximity_gap: f32) -> f32 {
+        // (w_rap + d - 1/2) * e_proximity_gap
+        let first_term = (self.n_trace_polys + self.d - 0.5) * e_proximity_gap;
+
+        // (2m + 1) * (|D| + 1) * FRI_FOLD * num_folding_rounds
+        // ----------------------------------------------------
+        //                 sqrt(ρ) * |K|
+        let second_term = {
+            let numerator = (2.0 * M + 1.0)
+                * (self.lde_domain_size + 1.0)
+                * (FRI_FOLD * self.num_folding_rounds) as f32;
+            let denominator = RHO.sqrt() * self.ext_field_size;
+            numerator / denominator
+        };
+
+        first_term + second_term
+    }
+
+    fn e_fri(&self, theta: f32, e_proximity_gap: f32) -> f32 {
+        let e_fri_constant = self.e_fri_constant(e_proximity_gap);
+
+        let e_fri_queries = e_fri_queries(theta);
+
+        e_fri_constant + e_fri_queries
+    }
+
+    fn e_ali(&self, l_plus: f32) -> f32 {
+        l_plus * self.n_trace_polys / self.ext_field_size
+    }
+
+    fn e_deep(&self, l_plus: f32) -> f32 {
+        let h_plus = self.trace_domain_size + self.biggest_combo;
+
+        let numerator = self.d * (h_plus - 1.0) + (self.trace_domain_size - 1.0);
+        let denominator = self.ext_field_size - self.trace_domain_size - self.lde_domain_size;
+        l_plus * numerator / denominator
+    }
+
+    fn e_deep_ali(&self, l_plus: f32) -> f32 {
+        self.e_deep(l_plus) + self.e_ali(l_plus)
+    }
+}
+
+#[cfg(test)]
+mod toy_model {
+    use super::*;
+    fn toy_model(coeffs_size: usize) {
+        // e, field extension degree
+        let ext_size = baby_bear::ExtElem::EXT_SIZE;
+        let field_size = baby_bear::P as f32;
+        let ext_field_size = libm::powf(field_size, ext_size as f32);
+        let trace_domain_size = (coeffs_size / ext_size) as f32;
+        let lde_domain_size = trace_domain_size * INV_RATE as f32;
+
+        let num_folding_rounds = num_folding_rounds(coeffs_size, ext_size);
+
+        let params = Params {
+            n_sigma_mem: 0,
```

Comment:
the main open question on my mind is about if/how to model the lookup/permutation argument in the toy problem analysis. 

would be nice to give some care to this question, as I think the question of whether or not to omit these arguments will be the difference between 97 bits and 100 bits of soundness.

Reply:
here's the numerical summary if we don't set n_sigma_mem and n_sigam_bytes to 0

<img width="489" alt="image" src="https://github.com/risc0/risc0/assets/15272444/b336b3c7-7fcb-4a31-8feb-81dcf38345ab">